packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/ButtonItem/buttonItem.js

```diff
@@ -43,30 +57,92 @@ const ButtonItem = props => {
         paddingLeft
     };
 
-    const cssButtonTypeSuffix =
-        buttonType.charAt(0).toUpperCase() + buttonType.substring(1);
+    const ref = useRef();
 
     let linkProps = {};
-    let LinkComponent = 'div';
+    let url = '';
     if (typeof link === 'string') {
         linkProps = resolveLinkProps(link, linkType);
-        LinkComponent = linkProps.to ? Link : 'a';
+        url = linkProps.to ? linkProps.to : linkProps.href;
+    }
+
+    const typeToPriorityMapping = {
+        primary: 'high',
+        secondary: 'normal',
+        link: 'low'
+    };
+
+    const [minWidth, setMinWidth] = useState(0);
+
+    useEffect(() => {
+        if (typeof renderCallback === 'function') {
+            renderCallback(ref.current.offsetWidth, minWidth => {
+                setMinWidth(minWidth);
+                ref.current.style.minWidth = minWidth + 'px';
```

Comment:
We should use React state to have `setMinWidth` update the element. This is as simple as assigning `minWidth` from `useState()` to the style property minWidth for that element.